tcp/spring-cloud-starter-stream-processor-tcp-client/src/main/java/org/springframework/cloud/stream/app/tcp/client/processor/TcpClientProcessorConfiguration.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.tcp.client.processor;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.app.tcp.EncoderDecoderFactoryBean;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.ip.config.TcpConnectionFactoryFactoryBean;
+import org.springframework.integration.ip.tcp.TcpReceivingChannelAdapter;
+import org.springframework.integration.ip.tcp.TcpSendingMessageHandler;
+import org.springframework.integration.ip.tcp.connection.AbstractConnectionFactory;
+import org.springframework.integration.ip.tcp.connection.TcpMessageMapper;
+import org.springframework.integration.ip.tcp.serializer.AbstractByteArraySerializer;
+
+/**
+ * A processor application that acts as a TCP client.
+ *
+ * @author Ilayaperumal Gopinathan
+ * @author Gary Russell
+ */
+@EnableBinding(Processor.class)
+@EnableConfigurationProperties(TcpClientProcessorProperties.class)
+public class TcpClientProcessorConfiguration {
+
+	@Autowired
+	private Processor channels;
+
+	@Autowired
+	private TcpClientProcessorProperties properties;
+
+	@Bean
+	public TcpReceivingChannelAdapter adapter(
+			@Qualifier("tcpClientConnectionFactory") AbstractConnectionFactory connectionFactory) {
+		TcpReceivingChannelAdapter adapter = new TcpReceivingChannelAdapter();
+		adapter.setConnectionFactory(connectionFactory);
+		adapter.setClientMode(true);
+		adapter.setRetryInterval(this.properties.getRetryInterval());
+		adapter.setOutputChannel(this.channels.output());
+		return adapter;
+	}
+
+	@Bean
+	@ServiceActivator(inputChannel= Processor.INPUT)
+	public TcpSendingMessageHandler sendingMessageHandler(
+			@Qualifier("tcpClientConnectionFactory") AbstractConnectionFactory connectionFactory) {
+		TcpSendingMessageHandler sendingMessageHandler = new TcpSendingMessageHandler();
+		sendingMessageHandler.setConnectionFactory(connectionFactory);
+		return sendingMessageHandler;
+	}
```

Comment:
The server does not have to send the payload back in a dumb way. As a
matter of fact, it can also be totally asynchronous to its input.
One use case is also to have the processor initiate the conversation with
some data, and then have the server be the driver.

Le 22 juil. 2016 9:22 PM, "Artem Bilan" <notifications@github.com> a écrit :

> In
> tcp/spring-cloud-starter-stream-processor-tcp-client/src/main/java/org/springframework/cloud/stream/app/tcp/client/processor/TcpClientProcessorConfiguration.java
> <https://github.com/spring-cloud/spring-cloud-stream-app-starters/pull/181#discussion_r71931861>
> :
>
> > +		TcpReceivingChannelAdapter adapter = new TcpReceivingChannelAdapter();
> > +		adapter.setConnectionFactory(connectionFactory);
> > +		adapter.setClientMode(true);
> > +		adapter.setRetryInterval(this.properties.getRetryInterval());
> > +		adapter.setOutputChannel(this.channels.output());
> > +		return adapter;
> > +	}
> > +
> > +	@Bean
> > +	@ServiceActivator(inputChannel= Processor.INPUT)
> > +	public TcpSendingMessageHandler sendingMessageHandler(
> > +			@Qualifier("tcpClientConnectionFactory") AbstractConnectionFactory connectionFactory) {
> > +		TcpSendingMessageHandler sendingMessageHandler = new TcpSendingMessageHandler();
> > +		sendingMessageHandler.setConnectionFactory(connectionFactory);
> > +		return sendingMessageHandler;
> > +	}
>
> I would like to know a *semantic payload* of such a processor.
> Looks like it's just shift incoming message over the network, twice (!):
> to TCP server and back.
> And doesn't do any business logic.
>
> Why something like simple ThreadPoolExecutor shifting isn't enough?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/spring-cloud/spring-cloud-stream-app-starters/pull/181/files/d0da7388f40614af86d416a337156367703d4932#r71931861>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATIluIACcjDtkNGkdJgfP3c_V7Fxiblks5qYRhYgaJpZM4JP07U>
> .
>